lib/google/serverless/exec.rb

```diff
@@ -814,7 +814,10 @@ def start_build_strategy app_info
           image = container ? container["image"] : image_from_build(app_info)
         else
           env_variables = {}
-          app_info["spec"]["template"]["spec"]["containers"][0]["env"].each { |env| env_variables[env["name"]] = env["value"] }
+          app_env = app_info["spec"]["template"]["spec"]["containers"][0]["env"]
+          if app_env
+            app_env.each { |env| env_variables[env["name"]] = env["value"] }
```

Comment:
Another Ruby tip: you can do this without the `if app_env` by saying:

```ruby
app_env&.each { |env| env_variables[env["name"]] = env["value"] }
```

The `&.` operator first checks if app_env is nil, and calls `each` only if it's not nil.